conan/internal/api/detect_api.py

```diff
@@ -342,6 +353,18 @@ def _gcc_compiler(compiler_exe="gcc"):
     except (Exception,):  # to disable broad-except
         return None, None
 
+def _intel_compiler(compiler_exe="icx"):
+    try:
+        ret, out = detect_runner("%s --version" % compiler_exe)
+        if ret != 0:
+            return None, None
+        compiler = "intel-cc"
+        installed_version = re.search(r"([0-9][0-9][0-9][0-9]+(\.[0-9])?)", out).group()
```

Comment:
```suggestion
        installed_version = re.search(r"(202[0-9]+(\.[0-9])?)", out).group()
```
I am happy simplifying the regex and hardcode the first digits ,like ``(202[0-9]+(\.[0-9])?)``. It might be a bit simpler and more robust, detecting only 202X years versions.